### PR TITLE
FOGL-2349: FogLAMP reset should cleanup the FOGLAMP_DATA

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-2349: FogLAMP reset should cleanup the FOGLAMP_DATA

### CODE REVIEW

File: scripts/foglamp [hunk 1]
```diff
@@ -164,6 +164,21 @@ foglamp_reset() {
...
+  find "$FOGLAMP_DATA/logs" -depth -type f -exec rm {} \;
+  echo "Removed user data in $FOGLAMP_DATA/logs"
+  # Remove user data: var
+  find "$FOGLAMP_DATA/var" -depth -type f -exec rm {} \;
+  echo "Removed user data in $FOGLAMP_DATA/var"
+  # Remove user data: extras, with one exclusion
+  find "$FOGLAMP_DATA/extras" -depth -type f -not -name fogbench_sensor_coap.template.json -exec rm {} \;
+  echo "Removed user data in $FOGLAMP_DATA/extras"
+  # Remove user etc: extras, with one exclusion
+  find "$FOGLAMP_DATA/etc" -depth -type f -not -name storage.json -exec rm {} \;
```
Comment (praveen-garg) [inline scripts/foglamp]: We need to keep default `etc/certs/` which are 
foglamp.key, foglamp.cert, 
ca.key, ca.cert, ca.csr,  user.key, user.cert, 
admin.key & admin.cert (created by scripts/certificates & scripts/auth_certificates)  but delete others.
Also it appears to me we need to retain  `etc/kerberos` directory structure with its README.rst but its contents i.e. .keytab files should be removed.

**Comment by praveen-garg** [comment]
@ashish-jabble Please have a look w.r.t FOGL-3424 and share your review and test feedback.

File: scripts/foglamp [hunk 2]
```diff
@@ -164,6 +164,25 @@ foglamp_reset() {
   # Remove user data: scripts
   rm -rf "$FOGLAMP_DATA/scripts"
   echo "Removed user data in $FOGLAMP_DATA/scripts"
```
Comment (ashish-jabble) [inline scripts/foglamp]: Better to say "from" ~in~

**Review by ashish-jabble — DISMISSED** [review]
We are very much concerned about the following directories cleanup as some REST API endpoints totally depends upon them.

```
backup
logs 
scripts
support
snapshots - plugins
```

**Review by praveen-garg — COMMENTED** [review]
@pintomax 
1) Also remove 
* $FOGLAMP_DATA/backup
* $FOGLAMP_DATA/snapshots
* $FOGLAMP_DATA/support
* $FOGLAMP_DATA/*.csv

File: scripts/foglamp (hunk 2 above)
Comment (praveen-garg) [reply scripts/foglamp]: Yes please, for all echo statements

**Review by praveen-garg — DISMISSED** [review]

File: scripts/foglamp [hunk 3]
```diff
@@ -163,7 +163,35 @@ foglamp_reset() {
 
   # Remove user data: scripts
   rm -rf "$FOGLAMP_DATA/scripts"
-  echo "Removed user data in $FOGLAMP_DATA/scripts"
+  echo "Removed user data from $FOGLAMP_DATA/scripts"
+  # Remove user data: logs
+  find "$FOGLAMP_DATA/logs" -depth -type f -exec rm {} \;
```
Comment (praveen-garg) [inline scripts/foglamp]: @pintomax will this remove directories under logs as well?

Comment (pintomax) [reply scripts/foglamp]: It currently removes files, directories under log are kept

Comment (praveen-garg) [reply scripts/foglamp]: We only need to preserve, what foglamp have when it starts. Let me find the directory tree structure,  We see, when (vanilla) foglamp starts.

Comment (pintomax) [reply scripts/foglamp]: If we need to remove child directories I'll add a change

File: scripts/foglamp [hunk 4]
```diff
@@ -163,7 +163,35 @@ foglamp_reset() {
...
+  # Remove user data: var
+  find "$FOGLAMP_DATA/var" -depth -type f -exec rm {} \;
+  echo "Removed user data from $FOGLAMP_DATA/var"
+  # Remove user data: extras, with one exclusion
+  find "$FOGLAMP_DATA/extras" -depth -type f -not -name fogbench_sensor_coap.template.json -exec rm {} \;
+  echo "Removed user data from $FOGLAMP_DATA/extras"
+  # Remove user etc/kerberos: extras, with one exclusion
+  find "$FOGLAMP_DATA/etc/kerberos" -depth -type f -not -name "README.rst" -exec rm {} \;
+  # Remove user etc/certs: extras, with exclusions
+  find "$FOGLAMP_DATA/etc/certs" -depth -type f -not -name user.* -not -name foglamp.* -not -name ca.* -not -name admin.* -exec rm {} \;
```
Comment (praveen-garg) [inline scripts/foglamp:179]: We need to keep `pem` and `json` directories under `$FOGLAMP_DATA/etc/certs`

File: scripts/foglamp (hunk 3 above)
Comment (praveen-garg) [reply scripts/foglamp]: pi@raspberrypi:/usr/local/foglamp/data $ tree
	.
	├── core.err
	├── etc
	│   ├── certs
	│   │   ├── admin.cert
	│   │   ├── admin.csr
	│   │   ├── admin.key
	│   │   ├── ca.cert
	│   │   ├── ca.key
	│   │   ├── foglamp.cert
	│   │   ├── foglamp.csr
	│   │   ├── foglamp.key
	│   │   ├── json
	│   │   ├── pem
	│   │   ├── user.cert
	│   │   ├── user.csr
	│   │   └── user.key
	│   ├── kerberos
	│   │   └── README.rst
	│   └── storage.json
	├── extras
	│   └── fogbench
	│       └── fogbench_sensor_coap.template.json
	├── foglamp.db
	├── foglamp.db-shm
	├── foglamp.db-wal
	├── logs
	└── var
	    └── run
	        └── foglamp.core.pid

	10 directories, 19 files


@pintomax  
1) We need to keep `pem` and `json` directories under `$FOGLAMP_DATA/etc/certs`
2) remove everthing under `$FOGLAMP_DATA/logs` so just rm -rf $FOGLAMP_DATA/logs/*

**Review by praveen-garg — APPROVED** [review]

**Comment by pintomax** [comment]
@praveen-garg Currently only files (with some exclusions) are removed from data/etc/certs and its subfolders.

So directory tree is preserved, only files are removed

**Pull request merged**